src/main/java/co/com/bancolombia/task/UpdateProjectTask.java

```diff
@@ -0,0 +1,123 @@
+  }
+
+  @TaskAction
+  public void updateProject() throws IOException {
+    logger.lifecycle("Clean Architecture plugin version: {}", Utils.getVersionPlugin());
+    logger.lifecycle(
+        "Dependencies to update: {}",
+        (dependencies.isEmpty() ? "all" : String.join(", ", dependencies)));
+
+    Release latestRelease = getLatestPluginVersion();
+    if (latestRelease != null) {
+      logger.lifecycle("Latest version: {}", latestRelease.getTagName());
+
+      updatePlugin(latestRelease.getTagName());
+    }
+    updateDependencies();
+
+    builder.persist();
+  }
+
+  private Release getLatestPluginVersion() throws IOException {
+    try {
+      return RestConsumer.callRequest(PLUGIN_RELEASES, Release[].class)[0];
+    } catch (Exception e) {
+      logger.lifecycle("\tx Can't update the plugin " + e.getMessage());
+      return null;
+    }
+  }
+
+  private DependencyRelease getDependencyReleases(String dependency) throws IOException {
+    try {
+      return RestConsumer.callRequest(getDependencyEndpoint(dependency), DependencyRelease.class);
+    } catch (NullPointerException e) {
+      logger.lifecycle("\tx Can't update this dependency " + dependency);
+      return null;
+    }
+  }
+
+  private String getDependencyEndpoint(String dependency) {
+    String[] id = dependency.split(":");
+    if (id.length >= 2) {
+      return DEPENDENCY_RELEASES.replaceFirst("%s", id[0]).replace("%s", id[1]);
+    }
+    throw new IllegalArgumentException(
+        dependency
+            + "is not a valid dependency usage: gradle u "
+            + "--dependency "
+            + "dependency.group:artifact");
+  }
+
+  private void updatePlugin(String lastRelease) throws IOException {
+
+    if (lastRelease.equals(Utils.getVersionPlugin())) {
+      logger.lifecycle("You are already using the latest version of the plugin");
+      return;
+    }
+    logger.lifecycle("Updating the plugin ");
+
+    if (builder.isKotlin()) {
+      builder.updateExpression(
+          "build.gradle.kts",
+          "(id\\(\"co.com.bancolombia.cleanArchitecture\"\\)\\s?version\\s?).+",
+          "$1\"" + lastRelease + "\"");
+      return;
+    }
+    builder.updateExpression(
+        "gradle.properties", "(systemProp.version\\s?=\\s?).+", "$1" + lastRelease);
+    builder.updateExpression(
+        "build.gradle", "(cleanArchitectureVersion\\s?=\\s?)'.+'", "$1'" + lastRelease + "'");
+
+    logger.lifecycle("Plugin updated");
+  }
+
+  private void updateDependencies() throws IOException {
+    logger.lifecycle("Updating dependencies");
+    List<String> gradleFiles = Utils.getAllFilesWithExtension(builder.isKotlin());
+
+    if (dependencies.isEmpty()) {
+      // find all dependencies
+      for (String gradleFile : gradleFiles) {
+        dependencies.addAll(builder.findExpressions(gradleFile, "['\"].+:.+:[^\\$].+['\"]"));
+      }
+    }
+    dependencies = dependencies.stream().distinct().collect(Collectors.toList());
```

Comment:
you should use a set

Reply:
thanks for the suggestions, I refactored it